pyasic/miners/backends/vnish.py

```diff
@@ -275,7 +275,16 @@ async def get_config(self) -> MinerConfig:
         return self.config
 
     async def set_power_limit(self, wattage: int) -> bool:
+        config = await self.get_config()
```

Comment:
I pulled in the changes from https://github.com/UpstreamData/pyasic/pull/246 to get the preset power values and only call `self.web.set_power_limit(wattage)` if `wattage` is a preset value.

If it's not a valid preset, it currently just returns `False`. Should it return a message, log it, or handle it some other way?

Reply:
The best option might be to just get the preset with the next lowest wattage and set that.  Not sure if that's expected behavior, and it's something that really shouldn't be handled by the firmware, but it's probably the cleanest way to ensure something gets done.

Reply:
I think that option makes sense. I updated the code to set power to highest preset that is <= wattage (https://github.com/UpstreamData/pyasic/pull/245/commits/c542a0ee1337cc2906a5f847c9e10894bb43a5f4)